### DISCUSSION TITLE
Renaming Buttons Web Interface

### DISCUSSION THREAD

**Discussion by spippolone** [arendst/Tasmota#18836] [Q&A]
Hello everyone, a few years ago you flashed some sonoffs with the tasmota Firmware now outdated .. Sonoff-Tasmota 6.5.0.8 by Theo Arends but working.
I would like now and I no longer remember how to change the names of the WEB interface or the name that is written at the top sonoff and the buttons where it is written ON/OFF1 ON/OFF2 ON/OFF3 ON/OFF4 I would like to abbreviate the names of the connected actuators. . can someone give me a tip without having to study everything again??
here is the interface screenshot https://i.postimg.cc/B6VDfCmb/Sonoff.jpg
THANK YOU !!!

**Comment by lalo-uy** [comment]
WebButton<x> Change the name of the toggle buttons of the WEB UI. This
command accepts spaces in the name

**Reply by spippolone** [reply]
Thank you but I have 21:03:19 RSL: stat/sonoff/RESULT = {"Command":"Unknown"}

Suppose than I would change the Button 4 and rename ON/OFF in Valve 1 
how is the complete command ?

**Reply by sfromis** [reply]
Do not type `<x>` but instead use the index of the button you want to change. This is the usual format to document Tasmota commands.

**Comment by spippolone** [comment]
Hi I'm back from the weekend now..
Sure I wrote 
21:45:36 CMD: WebButton1 Valve
21:45:36 RSL: stat/sonoff/RESULT = {"Command":"Unknown"}

**Reply by barbudor** [reply]
Command "WebButton" was added in Tasmota v8.1.0

**Reply by spippolone** [reply]
OK.. now is clear, I have Sonoff-Tasmota 6.5.0.8

**Reply by sfromis** [reply]
https://tasmota.github.io/docs/Upgrading/#upgrade-flow

**Reply by spippolone** [reply]
My tasmota is positioned in an awkward place, do you think if I update OTA by webserver I will have problems ?
I wouldn't want to have to unplug it or manually reset and reconfigure everything...
Thank you for your patience

**Reply by barbudor** [reply]
Following exactly the upgrade flow reduce the risk. But it may happen that Murphy or other weird thing can happen
So if you can live with the button as they are and can't risk to have the Sonoff unavailable, I would advise to not change anything

**Reply by spippolone** [reply]
You are right.. the last question... I don't remember .. after firmware upgrade I must reload the configuration.. ? 
Is it compatible with my old backup from  6.5.0.8(sonoff) firmware ??

**Reply by barbudor** [reply]
If you follow accurately the upgrade path, you shouldn't loose your configuration
However, it is not a bad idea to save the configuration before starting and after each step. Just in case.

![image](https://github.com/arendst/Tasmota/assets/2996042/ec1b6290-ea6a-4e44-bb1c-bffd8bf5dd01)

The blue version numbers there are links to the firmware that you MUST use and not any other one
- v6.7.1 : http://ota.tasmota.com/tasmota/release-6.7.1/sonoff.bin
- v7.2.0 : http://ota.tasmota.com/tasmota/release-7.2.0/tasmota-lite.bin
- v8.5.1 : http://ota.tasmota.com/tasmota/release-8.5.1/tasmota-lite.bin
- v9.1.0 : http://ota.tasmota.com/tasmota/release-9.1.0/tasmota-lite.bin.gz
- v12.5.0 : http://ota.tasmota.com/tasmota/release/tasmota-lite.bin.gz

Do not bother downloading the file.
If your tasmota has access to the Internet, directly use that URL in the "Upgrade by web server" box:

![image](https://github.com/arendst/Tasmota/assets/2996042/9a2de823-71ab-48bf-bdd7-47a8a5c5dfeb)

Before starting, insure that you have "SaveData 1"

If you have to download the file and use the "Upgrade by file upload", do not unzip the "bin.gz" files (v9.1.0 and above)
They are intended to be uploaded as-is (Tasmota will unzip automatically)

Never use a "-minimal" version but the "-lite" that are listed above

After each upgrade, check that the device is working well. eventually restart it once more before the next upgrade

**Reply by spippolone** [reply]
Thank you Barbudor and thank you to all, I solved !! 

![Sonoff](https://github.com/arendst/Tasmota/assets/39454504/b56aa6f3-e82a-4ca5-bbaf-fadab394db62)

**Reply by spippolone** [reply]
Now.. that I enjoyed it I would also like to do the other one, which has firmware
Sonoff-Tasmota 5.13.1 da Theo Arends
I tryed the same sequence, but if I go in Firmware update and put in http://ota.tasmota.com/tasmota/release-5.14.0/sonoff.bin
it show UPDATING STARTED but after reboot it don't answer I HAVE LOST IT

**Reply by sfromis** [reply]
According to the link I posted, upgrading from 5.13.1 **has** to go via 5.14.0 before going to 6.7.1 and so on. Not following the sequence strictly may cause significant trouble....

**Reply by spippolone** [reply]
In fact I noticed this, and I installed 5.14.0 but I lost the Tasmota, I think that now I must reinitialize it

**Reply by barbudor** [reply]
Unfortunately sometime you get hit by Murphys'

**Reply by spippolone** [reply]
Reload Firmware by Tasmotizer... and button name changed now is all OK thanks for your help !

**Comment by PrasadNikapitiya** [comment]
How to add Tasmota's toggle switch to Tasmota's user interfaces.,
